### PR TITLE
Refactoring commands to avoid user inputs

### PR DESCRIPTION
**Pull request by sudovickili**
### Tl;dr

image

1. This branch allows the bot to be configured without Privileged Gateway Intents and installed via a _very_ narrow set of permissions compared to the admin scope we were using before. It should be something like:
    https://discord.com/oauth2/authorize?client_id=935390564403261481&scope=applications.commands%20bot&permissions=268486720
2. This branch refactors `/starry token-rule add`, `/starry token-rule edit`, and `/starry token-rule remove` to use modals or select dropdowns instead of user inputs and emoji reactions.
3. The bot technically still supports emoji reactions if we want, I just didn't end up using them. User inputs definitely no longer work, as that's the one thing Discord wants us to stop doing.

### Description:

    Discord.js doesn't like letting bots having access to message
    contents unless absolutely necessary. Our installation wizard-like
    UX is not considered "necessary" by their standards, so we'll
    replace them with their recommendation interaction patterns
    instead: select dropdowns and modal forms.

If you want to look at this closer in the code, the changes have been grouped into independently reviewable commits, each of which have a paragraph or two of context.

### Suggested QA:

- [ ] Use `/starry farewell` to remove bot, ensure that created roles have been removed
- [ ] Re-add the bot, expect a welcome message
- [ ] Use `/starry token-rule add` normally
  - [ ] Select "Choose a token" normally
  - [ ] Select "I need to make a token" normally
  - [ ] Select the DAODAO option normally
- [ ] Use `/starry token-rule add` incorrectly, expect helpful error
  - [ ] For the first two options, enter invalid cw20 address
  - [ ] For DAODAO option, type in an incorrect URL
- [ ] Use `starry join` flow from an account that *doesn't* have the added token, expect no roles when finished
- [ ] Use `starry join` flow from an account that *does* have the added token, expect all applicable roles to be added

### Further QA:

- [ ] In the middle of a wizard, click on a button from another step, expect a helpful error or it to be ignored
- [ ] Reply to messages that are expecting emoji reaction inputs
- [ ] Kick starrybot (not farewell) and re-add, ensuring normal functioning

### CODE REVIEW

**Comment by sudovickili** [comment]
When this is merged, we can update the buttons on the main site + webapp to ask for the reduced permission:

- https://github.com/starryzone/cosmos-download-welcome-webapp/pull/7
- https://github.com/starryzone/cosmos-webapp/pull/10
